### ISSUE TITLE
Contact photos not showing

### ISSUE THREAD

**Issue by ygoe** [signalapp/Signal-Android#13827]
### Guidelines

- [X] I have searched searched open and closed issues for duplicates
- [X] I am submitting a bug report for existing functionality that does not work as intended
- [X] This isn't a feature request or a discussion topic

### Bug description

Some of my contacts do not show the photos. The photos are visible in Android's contacts app, but not in the Signal app. Some photos are still there but others not.

I had some trouble with DAVx to not properly sync contacts from my CardDAV server to Android. I have deleted that sync profile and recreated it. Now all contacts show up properly again. The contacts are visible everywhere but not in the Signal app.

I went to the "new chat" view and selected the "Update" command from the top menu. The circle was spinning for a moment and nothing has changed. I tried that several times and rebooted the phone in between.

Needless to say that the contact photos are also missing on the Windows desktop app then. There might be different issues preventing a proper upgrade there, but I'll get to that as soon as the Android app has proper data to sync.

### Device

Samsung S10e

### Android version

12

### Signal version

7.26.1

**Comment by ygoe** [comment]
Nothing has changed.

I'm wondering if it may help to somehow backup my data on the phone (never done that before), then uninstall the app and delete its data, and reinstall with a restored backup. Would that work to fix all broken data structures in the app?

**Comment by greyson-signal** [comment]
Just to be sure, have you enabled Settings > Chats > "Use address book photos"

**Comment by ygoe** [comment]
No, it was disabled. I can't remember touching that setting, didn't even know it exists.

I've enabled it now, also used the refresh command on the new chat contacts selection page, but so far nothing has changed about the photos.

**Comment by har-123** [comment]
Can you please share the recording of the issue. I will start looking into it.

**Comment by ygoe** [comment]
I have no recording.

**Comment by ygoe** [comment]
Still valid. Is anybody trying to care about issues faster than the configured bot expiration timeout? Should the bot be reconfigured to give the team more time to look at reports?

**Comment by johan-cb** [comment]
I can confirm this issue on Signal 7.41.4 on Android 15. Some of my contacts don't show pictures even though the contact has a picture set in contact cards.

**Comment by zecuse** [comment]
Adding my 2 cents. This occurred on my old phone which had Android 15 sometime late last year. I never bothered to look into fixing it as I knew I'd be getting a new phone this year. Well, a month has gone by since getting the new phone and the problem persists.

I've cleared the app's cache, changed the contact's photo, updated the contact's name to match what's displayed in Signal and it still refuses to use the address book photo again.

Android version: 15
Signal version: 7.42.2

**Comment by ygoe** [comment]
Still valid. Is anybody trying to care about issues faster than the configured bot expiration timeout? Should the bot be reconfigured to give the team more time to look at reports?

**Comment by zecuse** [comment]
> Adding my 2 cents. This occurred on my old phone which had Android 15 sometime late last year. I never bothered to look into fixing it as I knew I'd be getting a new phone this year. Well, a month has gone by since getting the new phone and the problem persists.
> 
> I've cleared the app's cache, changed the contact's photo, updated the contact's name to match what's displayed in Signal and it still refuses to use the address book photo again.
> 
> Android version: 15 Signal version: 7.42.2

Coming back to this, the problem still exists with the following observations:

- Android version 15
- Signal version 7.53.4
- This is only happening with my girlfriend's icon who is the only contact that I've used Signal with. She's on an iPhone if that detail matters.
- I do happen to see 2 other contact's when I try to test updating contact photos. These 2 contact's DO IN FACT change appropriately to a photo of my choosing and go back to their initials when I remove the assigned photo.
- This is NOT the case for my girlfriend. Her photo is stuck as the default, incognito looking icon Signal uses.
- It doesn't matter if I remove or reassign a photo to her. Only her icon is stuck.

**Comment by Joyescat** [comment]
My girlfriend had the same issue.
It happened because i had the option to hide my phone number for security reason.
My girlfriend added me on signal with my username. So my signal profile on her phone had no phone number.

To fix it : 
- i desactivated the setting to hide my phone number
- she looked for my phone number in the search menu
- she activated the setting to show contact photo instead of signal photo
- at first it didn't appear to be fixed but after few minute it worked 🎉🎉🎉

**Comment by ygoe** [comment]
I don't see these options. But looking around, I found another quirks.

I went to the details page of a contact that's lacking the photo. There's the menu entry "phone contact details". When I tap on that, I can select a contact app to open. I cancel by going back. Then after 2 seconds, the contact's photo appears at the top. But 2 seconds later, it disappears again and the initials come back.

Something's terribly broken in the app. It *can* show the photo, but apparently actively decides not to do. Is anybody actually fixing bugs here?

**Comment by psklb** [comment]
Hello, 
my experience on android signal app and desktop app with this problem (photo does't appear for some contacts) :
On th android apps : I notice that when I show contact details on contacts without photos, it was not all the details I see on my android contact
plus : the contact view shows 'signal contact'
So I think there is a "only singal" contact that overwrite my android contact. 
I show the contact details from signal app and delete it. 
Immediately, the contact become the android contact (without lose of conversation  (btu you can archive before to be sure))

On the destop app : params / conversations (chats in english ?) / import contacts 

Now everything is OK for me

**Comment by prettyGrace** [comment]
Registered and honest

**Comment by ygoe** [comment]
@psklb Not sure what you did or saw there but I cannot confirm it. First, all contacts, with or without photos, show "Signal contact" for me. Second, I can't delete a contact in the Signal Android app, there's no option for that. I could only delete a chat, or block or report a contact.

**Comment by psklb** [comment]
> Second, I can't delete a contact in the Signal Android app

sorry, that's what I remembered. I've just re-try (but my contacts are now fine) and probably the delete action went after I choose 'contact information from the telephone' in the signal app (or something like this, my phone speak french) . 
But I confirm the contact I saw there was not so detailed that the contact I built in my adressbook ;  that's why I had made the assumption that there was 2 contacts, the less complete overwrtiing the "real" one.
I regret not being able to try again. Hope this help though
Best regards

**Comment by ygoe** [comment]
Still relevant.

**Comment by ygoe** [comment]
Dear @stale bot, please also remember to remove your added "wontfix" label again. It looks so depressing. Thank you.

**Comment by ygoe** [comment]
Still relevant. Is anybody there?

**Comment by croweykid** [comment]
Same issue, but on iOS.

**Comment by johan-cb** [comment]
Still relevant

- Signal Android 8.0.4
- Samsung OneUI 8.5 - Android 16

**Comment by zdivelbiss** [comment]
I am having this issue as well.

Android 16, updated app.

**Comment by jenbrett818-lab** [comment]
Verify and update so

**Comment by ygoe** [comment]
Sorry, I have to declare Signal for Andorid unmaintained. Do I need to use WhatsApp again?

@greyson-signal @alex-signal You're the most active recent committers here. Can you please confirm that this issue has not fallen off the table? It's a bit annoying but with all the time, I got used to Signal not showing contact photos anymore. It's just a very primitive prototype app that doesn't try to be user-friendly. I'm ready to help debugging this, but I need guidance. I'm a .NET developer who has given up on Kotlin after the first try, and Java could never convince me. So no coding here.

**Comment by greyson-signal** [comment]
Hi there. Some things that initially come to mind for anyone experiencing the issue:

- Do you have Settings > Chats > "Use address book photos" enabled? If not, enable it.
- Are the affected contacts linked to a system contact from the app's perspective? If you tap their avatar in the chat list, do you see a "Phone contact info" row in the bottom sheet that pops up? If not, it could be the case that your contact turned off phone number discoverability (Settings > Privacy > Phone number > Who can find me by phone number). If that were to happen, then we disassociate their contact info.
- Was your current install seeded from a backup/restore? I believe there was an old bug where we would import contact links from a backup, which could result in us restoring broken contact links. We no longer do that, but I don't know if there's any remedy in the app (besides doing another backup/restore) if you already hit that issue.

I don't see any evidence that this is a widespread problem (i.e. the contact linking works fine for everyone that I know), but it's clearly a problem for some people, so hopefully we can figure it out.

**Comment by ygoe** [comment]
"Use address book photos" is enabled.

Affected and non-affected chats are both phone contacts and non-phone contacts. In short: There is no correlation here, every combination exists. But most chats are phone contacts here.

I don't even know how to to backups and restores on Android. I believe that the app isolation doesn't even allow that. Will be interesting though when I switch to a new phone maybe this year. Until then, this is a primary installation, quite old, though.

Interestingly, the missing contact photos can briefly show up after opening chats, viewing contact details and returning to the chat list. They may be visible for less than a second and disappear again right away. So the information is here, it's just not shown properly. This is not reproducible, though.

Maybe log the exact reason each time the chat list photo is re-rendered.

**Comment by zecuse** [comment]
> * Are the affected contacts linked to a system contact from the app's perspective? If you tap their avatar in the chat list, do you see a "Phone contact info" row in the bottom sheet that pops up? If not, it could be the case that your contact turned off phone number discoverability (Settings > Privacy > Phone number > Who can find me by phone number). If that were to happen, then we disassociate their contact info.

@greyson-signal As I mentioned earlier in this issue, I've been having this problem with my (now) fiancé who is the only person I use Signal with. This has persisted ever since it first appeared last year. By any chance, since she uses an iPhone, do you know if iPhones have phone number discoverability turned off by default?

<details>
<summary>Here's the screenshots of her profile when I click her avatar:</summary>
<img width="1080" height="1458" alt="Image" src="https://github.com/user-attachments/assets/d63bc810-ab88-4e7a-89ee-65e46d754e87" />

<img width="1080" height="1355" alt="Image" src="https://github.com/user-attachments/assets/85ad982e-dcf7-444c-8425-1d86acc6a1be" />
</details>

Is this what you mean when you talk about not seeing a "Phone contact info" row at the bottom?

Android version: 16
Signal version: 8.9.1

**Comment by greyson-signal** [comment]
@zecuse Yeah, so based on that screenshot, that is not linked to a contact on your phone. One possibility is that you don't have the number linked correctly, like maybe somehow the number in your contacts isn't correct, or it just doesn't match whatever number your fiance used to register. The other possibility is that your fiancee has phone number discoverability turned off. It should not be turned off by default on iPhone, but it's possible that she turned it on during registration and didn't realize it. You can have her check `Settings > Privacy > Phone number > Who can find me by phone number`. Of course, the last possibility is just that there's no misconfiguration, it's just some weird bug. If none of the other things apply, logs from you and your fiance would be helpful.

And congrats on the engagement :)

**Comment by ygoe** [comment]
I asked my contacts what their settings are. All of them have "Who can find me by phone number" set to "Everybody".

There is one contact who said he doesn't have a photo set. But I have set one in my phonebook. And that's the photo I can briefly see sometimes, but normally it's missing (only initials shown).

I've also set a local photo for another contact but can't see it in Signal. I don't know yet if he has one set in his Signal profile.

There is another contact of which I don't have a photo in my phonebook, but I see the person's Signal profile photo.

And there are 3 contacts where I can see the photo I have set in my phonebook.

All in all, that setting doesn't seem to have to do anything with the behaviour. Considering that the photo briefly appears and quickly disappears for one contact, I strongly believe this is an application rendering bug.

**Comment by clavelm** [comment]
I'm hiding my phone number on signal android, and so does an iphone person I exchange with.

With no phone number associated to that person, signal does not link to it's contact on the phone and there is no way to force it.
On iphone, it is possible to add a field in the phone contact to force the connection in signal, even without phone number.

I should be able:

- either to assign a picture to a contact in signal the same way I can set a name,
- or having a way to force the link with a contact from the phone.

The only way for **me** to have a picture for that person on my android is to ask **them** to set one on their iphone signal app.

**Comment by ygoe** [comment]
Should I give up on it? Create a hundred more of these issues? Assign an AI agent to do it?
